src/utils.rs

```diff
@@ -133,55 +133,58 @@ pub mod format {
   use crate::errors::ErrorUnexpectedInputSize;
   use std::convert::TryInto;
 
-  fn u8_to_bits_le(byte: u8) -> Vec<bool> {
-    let mut ret = Vec::new();
-    for i in 0..8 {
-      ret.push(2u8.pow(i as u32) & byte > 0);
-    }
-    ret
+  fn u8_to_bits_le(byte: u8) -> [bool; 8] {
+    [
+      2u8.pow(0) & byte != 0,
+      2u8.pow(1) & byte != 0,
+      2u8.pow(2) & byte != 0,
+      2u8.pow(3) & byte != 0,
+
+      2u8.pow(4) & byte != 0,
+      2u8.pow(5) & byte != 0,
+      2u8.pow(6) & byte != 0,
+      2u8.pow(7) & byte != 0,
+    ]
```

Comment:
hmm this should also work, and is more compact
```suggestion
    [0, 1, 2, 3, 4, 5, 6, 7]
      .map(|i| 2u8.pow(i) & byte != 0)
```

Reply:
nope that regresses performance massively